tao_compiler/mlir/disc/utils/cycle_detector.cc

```diff
@@ -338,4 +338,10 @@ std::vector<int32_t> GraphCycles::AllNodesInPostOrder() const {
   return all_nodes;
 }
 
+bool GraphCycles::IsActivateNode(int32_t node) const {
+  llvm::DenseSet<int32_t> free_nodes_set(rep_->free_nodes.begin(),
```

Comment:
why not just iterate over `rep_->free_nodes`? Constructing a DenseSet iterates it too ...

Reply:
Done.